torch/distributed/checkpoint/planner_helpers.py

```diff
@@ -185,8 +188,40 @@ def _create_default_metadata_only_plan(state_dict: STATE_DICT_TYPE) -> SavePlan:
     return SavePlan(requests)
 
 
+def _get_box_for(
+    tensor: DTensor, idx: Optional[int]
+) -> Tuple[torch.Size, torch.Size]:
+    device_mesh = tensor.device_mesh
+    assert device_mesh.ndim == 1, "Only 1D DeviceMeshes currently handled"
+
+    size = tensor.to_local().size()
+    offsets = tensor._spec.local_offsets
+
+    return (torch.Size(offsets), size)
+
+
+def _create_write_items_for_dtensor(fqn: str, tensor: DTensor) -> WriteItem:
+    offsets, sizes = _get_box_for(
+        tensor, tensor.device_mesh.get_coordinate_on_dim(0)
+    )
+    return WriteItem(
+        index=MetadataIndex(fqn, offsets),
+        type=WriteItemType.SHARD,
+        tensor_data=TensorWriteData(
+            chunk=ChunkStorageMetadata(
+                offsets=offsets,
+                sizes=sizes,
+            ),
+            properties=TensorProperties.create_from_tensor(tensor.to_local()),
```

Comment:
I think as part of ST deprecation, we should start moving away from using ST internal data structures. DCP should have its own data structure if needed.

Reply:
> I think as part of ST deprecation, we should start moving away from using ST internal data structures. DCP should have its own data structure if needed.

Same as above.